frontend/src/components/LineChartComponents/LineChart.tsx

```diff
@@ -0,0 +1,761 @@
+  );
+
+  // Effect to update data in series
+  useEffect(() => {
+    // Skip effect if chart is not initialized yet
+    if (!chart) return;
+    // Also skip if there is no scenario or compartment selected
+    if (selectedScenario === null || !selectedCompartment) return;
+
+    // Create empty map to match dates
+    const dataMap = new Map<string, {[key: string]: number}>();
+
+    if (lineChartData) {
+      lineChartData.forEach((serie) => {
+        const id = serie.serieId;
+        if (activeScenarios && activeScenarios.includes(Number(id))) {
+          serie.values.forEach((entry) => {
+            dataMap.set(entry.day, {...dataMap.get(entry.day), [id]: entry.value as number});
+          });
+        } else if (groupFilterList && typeof id === 'string' && id.startsWith('group-filter-')) {
+          const groupFilterName = serie.name;
+          Object.values(groupFilterList).forEach((groupFilter) => {
+            if (groupFilter.name === groupFilterName && groupFilter.isVisible)
+              serie.values.forEach((entry) => {
+                dataMap.set(entry.day, {...dataMap.get(entry.day), [groupFilterName]: entry.value as number});
+              });
+          });
+        } else if (serie.openValueYField) {
+          serie.values.forEach((entry) => {
+            dataMap.set(entry.day, {...dataMap.get(entry.day), [serie.valueYField]: (entry.value as number[])[1]});
+            dataMap.set(entry.day, {
+              ...dataMap.get(entry.day),
+              [String(serie.openValueYField)]: (entry.value as number[])[0],
+            });
+          });
+        }
+      });
+    }
+
+    // Sort map by date
+    const dataMapSorted = new Map(Array.from(dataMap).sort(([a], [b]) => String(a).localeCompare(b)));
+    const data = Array.from(dataMapSorted).map(([day, data]) => {
+      return {date: day, ...data};
+    });
+
+    // Put data into series
+    chart.series.each((series, i) => {
+      // Set-up data processors for first series (not needed for others since all use the same data)
+      if (i === 0) {
+        series.data.processor = DataProcessor.new(root as Root, {
+          // Define date fields and their format (incoming format from API)
+          dateFields: ['date'],
+          dateFormat: 'yyyy-MM-dd',
+        });
+      }
+      // Link each series to data
+      series.data.setAll(data);
+    });
+
+    // Set up HTML tooltip
+    const tooltipHTML = `
+        ${'' /* Current Date and selected compartment name */}
+        <strong>{date.formatDate("${
+          localization.overrides && localization.overrides['dateFormat']
+            ? customT(localization.overrides['dateFormat'])
+            : defaultT('dateFormat')
+        }")} (${
+          localization.overrides && localization.overrides[`compartment.${selectedCompartment}`]
+            ? customT(localization.overrides[`compartment.${selectedCompartment}`])
+            : defaultT(`compartment.${selectedCompartment}`)
+        })</strong>
```

Comment:
I don't like that we need the `selectedCompartment` as an input only for this small part of the code. Maybe we can rename `selectedCompartment` to `yAxisLabel`? Then the user can pass an already internationalized string and the default label could be EN: `"Value"` and DE: `"Wert"`.